atd-vzv/src/views/nav/SideMapControlDateRange.js

```diff
@@ -1,102 +1,226 @@
+    .reset-button {
+      position: relative;
+      top: 5px;
+      right: 2px;
+      width: 22px;
+      height: 22px;
+    }
+  `;
+
   return (
-    <StyledDateRangePicker>
-      <Button
-        onClick={() => setIsCalendarOpen(!isCalendarOpen)}
-        // Add some margin below button when calendar is open
-        className={`w-100 ${isCalendarOpen && "mb-1"}`}
-        color="info"
-      >
-        Choose Date Range
-      </Button>
-      {isCalendarOpen && (
-        // Autosize as recommended in InfiniteCalendar docs
-        <AutoSizer disableHeight>
-          {({ width }) => (
-            <InfiniteCalendar
-              Component={CalendarWithRange}
-              selected={convertToDatePickerDateFormat(date)}
-              onSelect={convertToSocrataDateFormat}
-              width={width}
-              height={225}
-              min={mapDataMinDate}
-              max={mapDataMaxDate}
-              minDate={mapDataMinDate}
-              maxDate={mapDataMaxDate}
-              theme={calendarTheme}
-              locale={{
-                headerFormat: "MMM Do"
-              }}
-              displayOptions={{
-                showTodayHelper: false
-              }}
-            />
-          )}
-        </AutoSizer>
-      )}
-    </StyledDateRangePicker>
+    <StyledButtonContainer className="pr-0 picker-outline w-100">
+      <DateRangePicker
+        startDateId="start_date" // PropTypes.string.isRequired,
+        endDateId="end_date" // PropTypes.string.isRequired,
+        startDate={start} // momentPropTypes.momentObj or null,
+        endDate={end} // momentPropTypes.momentObj or null,
+        onDatesChange={handleDateChange} // PropTypes.func.isRequired,
+        focusedInput={focused} // PropTypes.oneOf([START_DATE, END_DATE]) or null,
+        onFocusChange={focusedInput => setFocused(focusedInput)} // PropTypes.func.isRequired,
+        minDate={dataStartDate}
+        maxDate={dataEndDate}
+        renderCalendarInfo={() => isMobile() && renderCalendarInfo()} // Render custom close button on mobile
+        calendarInfoPosition="top" // Position custom close button
+        appendToBody // Allow calendar to pop out over SideDrawer and Map components
+        withFullScreenPortal={isMobile()} // Show full screen picker on mobile
+        small
+        renderMonthElement={renderMonthElement} // Render year picker
+        orientation={isMobile() ? "vertical" : "horizontal"} // More mobile friendly than horizontal
+        isOutsideRange={() => false} // Enable past dates
+        isDayBlocked={isOutsideDateLimits} // Grey out dates
+      />
+      {/* Reset button to restore default date range */}
+      <FontAwesomeIcon
+        className="reset-button"
+        icon={faTimesCircle}
+        color={colors.info}
+        onClick={() => {
+          setStart(dataStartDate);
+          setEnd(dataEndDate);
+        }}
+      />
+    </StyledButtonContainer>
```

Comment:
Sheesh, that's a lot of properties for the DateRangePicker. It's looking really good though :) 👍

Reply:
I agree about the properties! The number of combinations made implementing this a bit tricky and overwhelming!  😅